src/metabase/api/search.clj

```diff
@@ -405,21 +406,23 @@
 (s/defn ^:private search
   "Builds a search query that includes all the searchable entities and runs it"
   [search-ctx :- SearchContext]
-  (let [search-query      (full-search-query search-ctx)
-        _                 (log/tracef "Searching with query:\n%s\n%s"
-                                      (u/pprint-to-str search-query)
-                                      (mdb.query/format-sql (first (mdb.query/compile search-query))))
-        reducible-results (mdb.query/reducible-query search-query :max-rows search-config/*db-max-results*)
-        xf                (comp
-                           (map t2.realize/realize)
-                           (filter check-permissions-for-model)
-                           ;; MySQL returns `:bookmark` and `:archived` as `1` or `0` so convert those to boolean as
-                           ;; needed
-                           (map #(update % :bookmark api/bit->boolean))
-                           (map #(update % :archived api/bit->boolean))
-                           (map (partial scoring/score-and-result (:search-string search-ctx)))
-                           (filter #(pos? (:score %))))
-        total-results     (scoring/top-results reducible-results search-config/max-filtered-results xf)]
+  (let [search-query       (full-search-query search-ctx)
+        _                  (log/tracef "Searching with query:\n%s\n%s"
+                                       (u/pprint-to-str search-query)
+                                       (mdb.query/format-sql (first (mdb.query/compile search-query))))
+        to-toucan-instance (fn [o] (t2.instance/instance (search-config/model-to-db-model (:model o)) o))
```

Comment:
added this fn to turn object from reducible-query to a fully qualified toucan2/instance so `mi/can-read?` could dispatch correctly.

Reply:
@camsaul not sure if this idiomatic with toucan2? 

AFAIK, reducible-query can only have `modelable` for a single model?

Reply:
That is correct. I think this looks fine to me